docs/modules/ROOT/pages/erc1155.adoc

```diff
@@ -19,3 +19,129 @@ This approach leads to massive gas savings for projects that require multiple to
 Because all state is held in a single contract, it is possible to operate over multiple tokens in a single transaction very efficiently. The standard provides two functions, xref:api:token/ERC1155.adoc#IERC1155-balanceOfBatch-address---uint256---[`balanceOfBatch`] and xref:api:token/ERC1155.adoc#IERC1155-safeBatchTransferFrom-address-address-uint256---uint256---bytes-[`safeBatchTransferFrom`], that make querying multiple balances and transferring multiple tokens simpler and less gas-intensive.
 
 In the spirit of the standard, we've also included batch operations in the non-standard functions, such as xref:api:token/ERC1155.adoc#ERC1155-_mintBatch-address-uint256---uint256---bytes-[`_mintBatch`].
+
+== Constructing an ERC1155 Token Contract
+
+We'll use ERC1155 to track multiple items in our game, which will each have their own unique attributes. We mint all items to the deployer of the contract, we can transfer them to players. Players are free to keep their tokens or trade them with other people as they see fit, as they would any other asset on the blockchain!  
+
+For simplicity we will mint all items in the constructor but you could add minting functionality to the contract to mint on demand to players.
```

Comment:
Maybe this would look good in a `TIP:` block?

```suggestion
For simplicity we will mint all items in the constructor but you could add minting functionality to the contract to mint on demand to players.

TIP: For an overview of minting mechanisms check out xref:erc20-supply.adoc[Creating ERC20 Supply].
```